### DISCUSSION TITLE
Slackmap Volunteers

### DISCUSSION THREAD

**Discussion by Can-Sahin** [International-Slackline-Association/slackmap#7] [General]
If anybody is interested in maintaining slackmap and up for some tasks could you provide your contact details here in a comment.

We will use `issues` to keep track of what needs to be done: 
 - https://github.com/International-Slackline-Association/slackmap/issues
 - https://github.com/International-Slackline-Association/slackmap-ui/issues
 - https://github.com/International-Slackline-Association/slackline-data/issues

**Comment by jorgedelamora** [comment]
Hello! I would like to contribute to mantaining slackmap. 

I'm curious if there's anything I can do here, I would be more valuable mantaining the UI as I've mostly worked as a front end dev with React. 

But also interested in any backend stuff.

jorgedlmv@gmail.com

**Comment by jordanbmcconnell** [comment]
Hi im interested in helping where i can. I have full stack experience, mostly with backend. also pretty versed in devops side of things

jordanbobbymcconnell@gmail.com

**Comment by Dosbodoke** [comment]
I want to contribute to the community, I found slackmap super interesting as I am developing a similar mobile application since it uses react-native-maps for rendering a map, where Highline points are displayed.

Let's keep in touch and hope we can work together!

**Reply by Dosbodoke** [reply]
Think I need to put the email here right? juangabriel4699@gmail.com

**Comment by Can-Sahin** [comment]
I updated the text. You can see what needs to be done in issues and we can communicate from there on.

**Comment by Dosbodoke** [comment]
I found it quite tricky to start a development environment since it uses AWS for the API, I have never used it. Can you provide some documentation?

Created [this issue](https://github.com/International-Slackline-Association/slackmap-ui/issues/6) on the slackmap-ui but I think it deserves an issue on the backend too

**Reply by Can-Sahin** [reply]
I don't understand the 'tricky' part of the web app. It's how you locally run all the react applications mostly. Set up env variables(if you don't want to figure them out by examining the web app, then you can ask via email) and run the 'start' script. I added it to the readme of the repo also.

**Reply by Dosbodoke** [reply]
The `.env` configuration is tricky since I have to get API keys from many different places. If you think sharing these keys has no problem you can send them to [juangabriel4699@gmail.com ](mailto:juangabriel4699@gmail.com).

Otherwise, a more complete README of how to get them would be useful.

**Reply by Can-Sahin** [reply]
In open-source projects, you don't share keys with everybody unless there is no way for them to obtain. So you need to get your mapbox API key. Its free of charge and take 2 mins.  There are no other keys. Its basically base paths for URL and I will send them to you. 

Mapbox API tokens cannot be shared because its subject to pricing. With a mistake, you can charge my credit card thousands of dollars you know. You can also find Mapbox API tokens online if you look for them but I cannot tell how ;)

**Comment by Icosaplex** [comment]
I'm interested in adding PWA support as per [this open issue](https://github.com/International-Slackline-Association/slackmap-ui/issues/3). I haven't done this in react before but have done it in angular and would like to see how different the implementation is.

theoscottvanzyl@gmail.com
